compiler-cli/src/dependencies.rs

```
@@ -672,6 +672,66 @@ fn is_same_requirements(
         }
     }
 
+    // For path dependencies, we need to check if their manifest.toml files have changed,
+    // which indicates that their dependencies have changed
+    for (key, requirement2) in requirements2 {
+        if let Requirement::Path { path } = requirement2 {
+            let dep_manifest_path = root_path.join(path).join("manifest.toml");
+
+            // Check if we have a cached hash for this path dependency's manifest.toml
+            // Use the dependency name instead of the path for the manifest hash file
+            let cached_hash_path = root_path
+                .join("build")
+                .join("packages")
+                .join(format!("{}.manifest_hash", key));
+
+            if !dep_manifest_path.exists() {
+                tracing::debug!("path_dependency_manifest_not_found_forcing_rebuild");
+                return Ok(false);
+            }
+
+            let manifest_content = match fs::read(&dep_manifest_path) {
+                Ok(content) => content,
+                Err(_) => {
+                    tracing::debug!("cannot_read_path_dependency_manifest_forcing_rebuild");
+                    return Ok(false);
+                }
+            };
+
+            let mut hasher = std::hash::DefaultHasher::new();
+            std::hash::Hash::hash(&manifest_content, &mut hasher);
+            let current_hash = std::hash::Hasher::finish(&hasher).to_string();
+
+            // If cached hash file doesn't exist, this is the first time we're checking this dependency
+            if !cached_hash_path.exists() {
+                // Save the current hash for future comparisons
+                if let Err(e) = fs::write(&cached_hash_path, &current_hash) {
```

The errors in this code are being silently discarded! They must always be passed back up.

---

Done!